index.bs

```diff
@@ -676,39 +671,19 @@ partial interface MLContext {
                 1. |graph|.{{MLGraph/[[inputDescriptors]]}}[|key|] must exist.
                 1. Let |inputDesc| be |graph|.{{MLGraph/[[inputDescriptors]]}}[|key|].
                 1. Let |inputSize| be 1.
-                1. If |value| is an {{MLArrayInput}}, then:
-                    1. The length of |value|.{{MLArrayInput/dimensions}} must be the same as the length of |inputDesc|.{{MLOperandDescriptor/dimensions}}.
-                    1. Let |i| be 0.
-                    1. While true:
-                        1. Let |dimension| be |value|.{{MLArrayInput/dimensions}}[|i|].
-                        1. |dimension| must be greater than 0.
-                        1. If |inputDesc|.{{MLOperandDescriptor/dimensions}}[|i|] is greater than 0, then |dimension| must be equal to |inputDesc|.{{MLOperandDescriptor/dimensions}}[|i|].
-                        1. Set |inputSize| to the product of |inputSize| and |dimension|.
-                        1. Increment |i| by 1.
-                        1. If |i| if equal to the length of |value|.{{MLArrayInput/dimensions}}, then break.
-                1. Else:
-                    1. For each |dimension| of |inputDesc|.{{MLOperandDescriptor/dimensions}}:
-                        1. The value of |dimension| must be greater than 0.
-                        1. Set |inputSize| to the product of |inputSize| and |dimension|.
-                1. If |value| is an {{MLArrayInput}}, then let |resource| be |value|.{{MLArrayInput/resource}}.
-                1. If |value| is an {{ArrayBufferView}}, then let |resource| be |value|.
-                1. If |resource| is an {{ArrayBufferView}}, then:
-                    1. The kind of |resource| must be compatible with |inputDesc|.{{MLOperandDescriptor/type}} according to [this table](#appendices-mloperandtype-arraybufferview-compatibility).
-                    1. The length of |resource| must be the same as |inputSize|.
+                1. For each |dimension| of |inputDesc|.{{MLOperandDescriptor/dimensions}}:
+                    1. The value of |dimension| must be greater than 0.
+                    1. Set |inputSize| to the product of |inputSize| and |dimension|.
+                1. The kind of |value| must be compatible with |inputDesc|.{{MLOperandDescriptor/type}} according to [this table](#appendices-mloperandtype-arraybufferview-compatibility).
+                1. The length of |value| must be the same as |inputSize|.
```

Comment:
This is element length. The algorithm steps above compute the element length (|inputSize|) of |input|. So the element length of |value| is used accordingly.

Probably using "element length" instead of "length" of |value| would be clearer?

Reply:
We could add our own definition of "length" (or give it a more specific name such as "element length") into this spec and then refer to that definition from within this spec.

Examples:
https://infra.spec.whatwg.org/#string-length
https://infra.spec.whatwg.org/#byte-sequence-length
https://infra.spec.whatwg.org/#string-code-point-length
https://dom.spec.whatwg.org/#concept-node-length
https://streams.spec.whatwg.org/#pull-into-descriptor-byte-length

Reply:
Adding our own definition of "element length" (or simply "length") works.

We can say something like:

```
In the rest of document, <dfn>length</dfn> of an ArrayBufferView is:

- array.[[ArrayLength]] for TypedArray (link to TypedArray spec in JavaScript)
- ... for XXX

<dfn>byteLength</def> of an ArrayBufferView is:

- ... for TypedArray
- ... for XXX
```

Then use bikeshed's auto-link `[=length=]` to refer to these definitions, so there's no ambiguity.

Reply:
Thanks for the pointers.

After revisiting ArrayBufferView [definition](https://webidl.spec.whatwg.org/#ArrayBufferView) and [usage](https://webidl.spec.whatwg.org/#buffersource-byte-length) in WebIDL spec, it looks like we'd better to use [[ByteLength]] for its viewing ArrayBuffer. That's actually raised by @wacky6 in his first comment.

So in the latest commit, I implemented the algorithm steps to calculate byte length of an MLOperandDescriptor. It then can be reused to validate the input and output ArrayBufferView's [[ByteLength].

Please take another look.